src/main/java/frc/robot/subsystems/Pigeon.java

```diff
@@ -0,0 +1,112 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import com.ctre.phoenix6.hardware.Pigeon2;
+import com.ctre.phoenix6.configs.Pigeon2Configuration;
+import frc.robot.util.preferenceconstants.DoublePreferenceConstant;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class Pigeon extends SubsystemBase {
+    private DoublePreferenceConstant p_PigeonYawCalibration;
+    private DoublePreferenceConstant p_PigeonRollCalibration; 
+    private DoublePreferenceConstant p_PigeonPitchCalibration;
+    private DoublePreferenceConstant p_GyroXSensitivityScalar;
+    private DoublePreferenceConstant p_GyroYSensitivityScalar;
+    private DoublePreferenceConstant p_GyroZSensitivityScalar;
+    private final double GRAVITY = 9.81;
+    private Pigeon2 base;
+    private int deviceID;
+    
+    public Pigeon(int deviceID, String canbus) {
+        this.deviceID = deviceID;
+        p_PigeonYawCalibration = new DoublePreferenceConstant("Pigeon" + deviceID + "/InitialYaw", 0);
+        p_PigeonRollCalibration = new DoublePreferenceConstant("Pigeon" + deviceID + "/InitialRoll", 0);
+        p_PigeonPitchCalibration = new DoublePreferenceConstant("Pigeon" + deviceID + "/InitialPitch", 0);
+        p_GyroXSensitivityScalar = new DoublePreferenceConstant("Pigeon" + deviceID + "/GyroXSensitivity", 0);
+        p_GyroYSensitivityScalar = new DoublePreferenceConstant("Pigeon" + deviceID + "/GyroYSensitivity", 0);
+        p_GyroZSensitivityScalar = new DoublePreferenceConstant("Pigeon" + deviceID + "/GyroZSensitivity", 0);
+        base = new Pigeon2(deviceID, canbus);
+        //configurePigeon();
+    }
+
+    public Pigeon(int deviceID) {
+        this(deviceID, "rio");
```

Comment:
I would have "rio" be a class constant. I assume this value is set in some phoenix configuration.
```
private final String DEFAULT_CAN_BUS = "rio";
```